client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/test/index.js

```diff
@@ -0,0 +1,391 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+import { recordEvent } from 'lib/tracks';
+import { ExternalLink, Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+jest.mock( '../../wcs-api.js' );
+import { acceptWcsTos, getWcsAssets } from '../../wcs-api.js';
+
+acceptWcsTos.mockReturnValue( Promise.resolve() );
+const wcsAssetsMock = {};
+getWcsAssets.mockReturnValue( Promise.resolve( wcsAssetsMock ) );
+
+import { ShippingBanner } from '../index.js';
+
+jest.mock( 'lib/tracks' );
+jest.mock( '@woocommerce/wc-admin-settings' );
+
+describe( 'Tracking impression in shippingBanner', () => {
+	const expectedTrackingData = {
+		banner_name: 'wcadmin_install_wcs_prompt',
+		jetpack_connected: true,
+		jetpack_installed: true,
+		wcs_installed: true,
+	};
+
+	const isJetpackConnected = true;
+	const activePlugins = {
+		includes: jest.fn().mockReturnValue( true ),
+	};
+
+	beforeEach( () => {
+		shallow(
+			<ShippingBanner
+				isJetpackConnected={ isJetpackConnected }
+				activePlugins={ activePlugins }
+				activatedPlugins={ [] }
+				installedPlugins={ [] }
+				wcsPluginSlug={ 'woocommerce-services' }
+				activationErrors={ [] }
+				installationErrors={ [] }
+			/>
+		);
+	} );
+
+	it( 'should record an event when user sees banner loaded', () => {
```

Comment:
Related to my comment above around the impression tracking in `componentDidMount` depending on the needs of that track - i.e. if we only want it recorded if the banner is visible, maybe we can add a test to verify that here.